assignment/week02/assign2/index.js

```diff
@@ -0,0 +1,207 @@
+
+c_in.addEventListener("change",function(event){
+    inputCategory=event.target.value;
+})
+c_out.addEventListener("change",function(event){
+    inputCategory=event.target.value;
+})
+m_input.addEventListener("input",function(event){
+    if(isNaN(event.target.value)){
+        window.alert("숫자 외의 문자는 입력할 수 없습니다.");
+        // 숫자아닌건 없애도록 replace하는 정규식
+        event.target.value=event.target.value.replace(/[^0-9]/g,"");
+    }else{
+        inputAmount=event.target.value;
+    }
+})
+m_content.addEventListener("change",function(event){
+    inputContent=event.target.value;
+})
+
+
+function infilter(){
+    clickedIncome=(!clickedIncome);
+    doFilter()
+}
+
+function outfilter(){
+    clickedOutcome=(!clickedOutcome);
+    doFilter()
+}
+
+function doFilter(){
+    detaillist.replaceChildren();
+    if(clickedIncome && clickedOutcome)
+        filteredList=history_list;
+    else if(clickedIncome)
+        filteredList=history_list.filter((each)=>each.amount>0);
+    else if(clickedOutcome)
+        filteredList=history_list.filter((each)=>each.amount<0);
+    else
+        filteredList=[];    
+
+    makeHistoryList();
+}
+
+function makeHistoryList(){
+    filteredList.forEach((each)=>{
+        
+        const eachlist = document.createElement('div');
+        const category = document.createElement('div');
+        const detail = document.createElement('div');
+        const amount = document.createElement('div');
+        const xbtn = document.createElement('div');
+        
+        category.innerText=each.category;
+        detail.innerText=each.detail;
+        amount.innerText=each.amount;
+        xbtn.innerText='x';
+        
+        eachlist.appendChild(category);
+        eachlist.appendChild(detail);
+        eachlist.appendChild(amount);
+        eachlist.appendChild(xbtn);
+    
+        eachlist.classList.add('eachlist');
+        detaillist.appendChild(eachlist);
+    
+        each.amount<0 ? amount.classList.add('out') : amount.classList.add('in')
+
+        xbtn.classList.add('xbtn');
+        xbtn.addEventListener('click', deleteHistory);
+
+
+    })
+}
+
+function deleteHistory(event){
+    closeModal.style.display="block";
+    let yes = document.querySelector('#closeYes');
+    let no = document.querySelector('#closeNo');
+    
+    yes.addEventListener("click",(e)=>{
+        if(e.target.value){
+            detaillist.removeChild(event.target.parentNode);
+            const idx= history_list.findIndex(each => each.detail ===event.target.previousElementSibling.previousElementSibling.innerText)
+            history_list.splice(idx,1);
+            closeModal.style.display="none";
+            doFilter()
+            calculateHistory();
+        }
+    })
+
+    no.addEventListener("click",(e)=>{
+        e.target.value && (closeModal.style.display="none")
+    })
+    
+}   
+
+function calculateHistory(){
+    let init_balance=0;
+    let income_total=0;
+    let outcome_total=0;
```

Comment:
여기도 magic number, 수치값 등을 나타내는 상수는 대문자로 바꿔주면 더 좋을것 같습니당!
```suggestion
    let INIT_BALANCE=0;
    let INCOME_TOTAL=0;
    let OUTCOME_TOTAL=0;
```

이런식으로 나타내면 단순 변수인지, 상수화해서 나타낸 값인지 한 눈에 비교하기 쉽습니댱